### DISCUSSION TITLE
Import with indexing turned OFF

### DISCUSSION THREAD

**Discussion by tilusnet** [photoprism/photoprism#4324] [Q&A]
Hello,

I am running Photoprism on a Pi 4, mapping in encrypted paths.
Since file I/O with encryption is enough load for the system during Import, especially with videos, I was wondering if there's a setting for turning Indexing OFF entirely while Import is running.
 
I would then run Indexing later, once the Originals path is up to date.

Can I turn indexing off in a Photoprism session?
I tried with `PHOTOPRISM_INDEX_WORKERS` set to 0 but it didn't work.

Thanks for your help!

**Comment by lastzero** [comment]
You cannot disable indexing when importing files, as imported files must be indexed to  become visible in the UI. However, importing should not index any additional files unless you manually trigger it or upload files to the originals folder via WebDAV at the same time.

**Comment by tilusnet** [comment]
Got it, yeah I assumed automatic indexing of the imported is by design to keep up with UI consistency - it's probably what's expected 90%+ of the time.

I was actually thinking of kick starting import (+dedupe+canonisation) as a separate standalone "headless" process, ie. no UI, just a `photoprism import ...` command line execution with docker.

Would that be a use case of wider use?

With that (database) indexing wouldn't be essential, as I only need my /originals readily deduped, canonised and /storage up to date with thumbnails, yaml files, etc -- long story short, the bulk of I/O done separately from indexing.
